### PR TITLE
Adding "[object NodeList]" === Object.prototype.toString.call(iter) t…

### LINKED ISSUE #9993: In IE11, NodeLists cannot be spread.
## Bug Report
Cause of Issue is known, will do a pull request.

**Current Behavior**
In IE11, NodeLists cannot be spread.
"Invalid attempt to spread non-iterable instance"

**Input Code**
```js
[...document.querySelectorAll('.filter-item')]
```

**Expected behavior/code**
The NodeList is turned into an Array.

**Babel Configuration (.babelrc, package.json, cli command)**

```js
{
  "presets": ["@babel/preset-react", "@babel/preset-env"]
}
```
```js
{
"devDependencies": {
    "@babel/core": "^7.2.2",
    "@babel/polyfill": "^7.2.5",
    "@babel/preset-env": "^7.2.3",
    "@babel/preset-react": "^7.0.0",
    "babel-core": "^7.0.0-bridge.0",
    "babel-jest": "^23.6.0",
    "babel-loader": "^8.0.5",
    "babel-preset-vue": "^2.0.2"
}
"dependencies": {
    "babel-polyfill": "^6.26.0"
}
}
```

**Environment**
- Babel version(s): see above
- Node/npm version: node v10.14.1 / npm 6.4.1
- OS: Ubuntu 18.04
- Monorepo: no
- How you are using Babel: loader

**Possible Solution**
In _toConsumableArray, add `"[object NodeList]" === Object.prototype.toString.call(iter)` additionally to `"[object Arguments]" === Object.prototype.toString.call(iter)`. In Chrome, the method will return `[object Arguments]`, but in IE11 (Windows 8.1) it will return `[object NodeList]`.
_Edit: removed redundant quotes_

### PR DESCRIPTION
**Pull request by ZainlessBrombie**
```
Adding "[object NodeList]" === Object.prototype.toString.call(iter) to check if object is eligible for Array.from() call for _toConsumableArray(arr). In IE11, the object returns the former string rather than "[object Arguments]" as already covered.
```

| Q                        | A
| ------------------------ | ---
| Fixed Issues?            | Fixes #9993 
| Patch: Bug Fix?          | Yes
| Major: Breaking Change?  | No
| Minor: New Feature?      | No
| Tests Added + Pass?      | Yes
| Any Dependency Changes?  | No
| License                  | MIT

When using the spread operator on an object to use as an array, the generated code passes the object to `_toConsumableArray()`. The function _toConsumableArray then decides how to handle the object, passing it to helper functions like `_iterableToArray(iter)`. One of the checks relies on `Object.prototype.toString.call(iter) === "[object Arguments]"`, which works in chrome. However in IE11 in Windows 8.1, `[object NodeList]` is returned instead and an exception is thrown, despite the object being processable by Array.from (or the replacement thereof). Since NodeLists are processable by Array.from, they should also be added to _iterableToArray.

edit: commit message put in code block

### CODE REVIEW

**Comment by ZainlessBrombie** [comment]
Optimally, the object should have contained Symbol.iterable. I am still investigating as to why the Symbols mismatch.

**Comment by ZainlessBrombie** [comment]
I have found the issue: The project is a vue project. Someone included the default cdn-version of google recaptcha so they could use it in vue instead of using the vue variant. The default version has its own polyfill and overwrites the main polyfill eventually. This is causing the iterator to be using a different key. I don't know why recaptcha overwrites without warning or even has global namespace pollution, but that is another matter entirely. The proposed "fix" here _may_ be included for robustnesses sake, but if that is necessary is questionable as other methods would surely have been affected as well, leaving more potential errors lingering.
TLDR; A script tag referencing google recaptcha overwrote the iterator. _Might_ want to merge the pull request for robustness, but it would only come to use in scenarios in which other components would likely be affected too.

**Comment by loganfsmyth** [comment]
Unfortunately I'm not sure this is something we'd want to land. There are many types that could be iterable but aren't without the polyfill, and making them work is something we leave up to polyfills because it would be unrealistic to support them otherwise, and special-casing `NodeList` wouldn't set a precedent that I don't think we'd want to encourage.

**Comment by ZainlessBrombie** [comment]
@loganfsmyth I absolutely agree! :)
Just wanted to offer the possibility of doing that, but it seems like a step sideways when we want to go forward, if that makes sense. No damage, but we shouldn't keep doing it so lets not start. Anyways, thank you for helping me out by pointing me to the code file, have a wonderful day!
